fenix/app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt

```diff
@@ -156,6 +157,14 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
         )
 
         context.components.useCases.searchUseCases.selectSearchEngine(engine)
+
+        val searchEngineName = when (engine.type) {
+            SearchEngine.Type.BUNDLED, SearchEngine.Type.BUNDLED_ADDITIONAL -> engine.name
+            SearchEngine.Type.CUSTOM -> "Custom"
+            SearchEngine.Type.APPLICATION -> "Application"
+        }
```

Comment:
Is this intentionally deviating from this same metric in `SearchDialogController`?

https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt#241-251

Reply:
Forgot we did that as well.  That's a good idea, I'll make that available as a extension.  Thanks!

Reply:
@MozillaNoah Please let me know if this is what you mean.  I've moved the code you linked to the extension so we can use it for this change as well.  Thanks

Reply:
The refactor looks great! :shipit: